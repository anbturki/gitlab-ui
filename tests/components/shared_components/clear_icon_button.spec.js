import { shallowMount } from '@vue/test-utils';
import ClearIconButton from '../../../src/components/shared_components/clear_icon_button/clear_icon_button.vue';

describe('Clear Icon Button', () => {
  let wrapper;

  const defaultPropsData = {
    title: 'Tooltip Text',
    tooltipContainer: '#divId',
  };

  const createComponent = (propsData = defaultPropsData) => {
    wrapper = shallowMount(ClearIconButton, { propsData });
  };

  beforeEach(() => {
    createComponent();
  });

  afterEach(() => {
    wrapper.destroy();
  });

  it('renders successfully', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
