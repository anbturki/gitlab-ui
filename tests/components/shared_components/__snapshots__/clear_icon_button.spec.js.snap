// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Clear Icon Button renders successfully 1`] = `
<button
  aria-hidden="true"
  class="gl-clear-icon-button"
  name="clear"
  title="Tooltip Text"
>
  <gl-icon-stub
    name="clear"
    size="16"
  />
</button>
`;
